sa/model.go

```diff
@@ -12,23 +12,156 @@ import (
 	"github.com/letsencrypt/boulder/core"
 	"github.com/letsencrypt/boulder/features"
 	"github.com/letsencrypt/boulder/probs"
+	"github.com/letsencrypt/boulder/revocation"
 )
 
-const (
-	regV1Fields        string = "id, jwk, jwk_sha256, contact, agreement, initialIP, createdAt, LockCol"
-	regV2Fields        string = regV1Fields + ", status"
-	pendingAuthzFields string = "id, identifier, registrationID, status, expires, combinations, LockCol"
-	authzFields        string = "id, identifier, registrationID, status, expires, combinations"
-	sctFields          string = "id, sctVersion, logID, timestamp, extensions, signature, certificateSerial, LockCol"
+// A `dbOneSelector` is anything that provides a `SelectOne` function.
+type dbOneSelector interface {
+	SelectOne(interface{}, string, ...interface{}) error
+}
 
-	// CertificateFields and CertificateStatusFields are also used by cert-checker and ocsp-updater
-	CertificateFields       string = "registrationID, serial, digest, der, issued, expires"
-	CertificateStatusFields string = "serial, subscriberApproved, status, ocspLastUpdated, revokedDate, revokedReason, lastExpirationNagSent, ocspResponse, LockCol"
+// A `dbSelector` is anything that provides a `Select` function.
+type dbSelector interface {
+	Select(interface{}, string, ...interface{}) ([]interface{}, error)
+}
 
-	// CertificateStatusFieldsv2 is used when the CertStatusOptimizationsMigrated
-	// feature flag is enabled and includes "notAfter" and "isExpired" fields
-	CertificateStatusFieldsv2 string = CertificateStatusFields + ", notAfter, isExpired"
-)
+const regFields = "id, jwk, jwk_sha256, contact, agreement, initialIP, createdAt, LockCol"
+const regFieldsv2 = regFields + ", status"
+
+// Select all fields of one registration model
+func SelectRegistration(s dbOneSelector, q string, args ...interface{}) (*regModelv1, error) {
```

Comment:
It looks like only `SelectCertificate`, `SelectCertificates`, `SelectCertificateStatuses`, and `SelectCertificateStatusesv2` need to be exported, is there a reason you've exported all of the Select* methods?